Jenkinsfile.windows

```diff
@@ -0,0 +1,281 @@
+                        "set \"rstudio_version_patch=${RSTUDIO_VERSION_PATCH}\" &&" +
+                        "set \"rstudio_version_suffix=${RSTUDIO_VERSION_SUFFIX}\" &&" +
+                        "set \"package_os=windows\" &&" +
+                        "make-package.bat clean ${FLAVOR} &&" +
+                        "cd ../.."
+                    }
+                  }
+
+                  stage('Tests') {
+                    steps {
+                      bat 'cd package/win32/build/src/cpp && rstudio-tests.bat --scope core'
+                    }
+                  }
+
+                  stage('Electron Tests') {
+                    when { environment name: 'FLAVOR', value: 'Electron' }
+                    steps {
+                      bat 'cd src/node/desktop && scripts\\run-unit-tests.cmd'
+                    }
+                  }
+
+                  stage('Sign, Upload, and Publish') {
+                    when { expression { return params.PUBLISH } }
+
+                    environment {
+                      PACKAGE_VERSION = "${RSTUDIO_RELEASE}".replace('+', '-')
+                    }
+
+                    stages {
+                      stage('Sign') {
+                        environment {
+                          PFX_FILE = credentials('ide-windows-signing-pfx') 
+                          PFX_PASS = credentials('ide-pfx-passphrase')
+                        }
+
+                        steps {
+                          script {
+                            def packageName = "RStudio-${PACKAGE_VERSION}-RelWithDebInfo"
+
+                              bat "\"C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.19041.0\\x86\\signtool\" sign /f %pfx-file% /p %PFX_FILE% %PFX_PASS% /v /debug /n \"RStudio PBC\" /t http://timestamp.digicert.com  package\\win32\\build\\${packageName}.exe"
+                              bat "\"C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.19041.0\\x86\\signtool\" verify /v /pa package\\win32\\build\\${packageName}.exe"
+                          }
+                        }
+                      }
+
+                      stage('Upload') {
+                        steps {
+                          script {
+                            def buildDest = "s3://rstudio-ide-build/${current_container.flavor}/windows"
+                            def packageName = "RStudio-${PACKAGE_VERSION}"
+
+                            // strip unhelpful suffixes from filenames
+                            bat "move package\\win32\\build\\${packageName}-RelWithDebInfo.exe package\\win32\\build\\${packageName}.exe"
+                            bat "move package\\win32\\build\\${packageName}-RelWithDebInfo.zip package\\win32\\build\\${packageName}.zip"
+
+                            // windows docker container cannot reach instance-metadata endpoint. supply credentials at upload.
+                            withAWS(role: 'jenkins', roleAccount: AWS_ACCOUNT_ID) {
```

Comment:
[Josh asked us ](https://github.com/rstudio/rstudio-pro/issues/3022)to start using role: `ide-build` instead of `jenkins-aws`. Any idea why this one is different?

Reply:
Is that a drop-in replacement, or is that going to break things?